src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IAsyncEnumerableConverterFactory.cs

```diff
@@ -15,6 +16,7 @@ internal sealed class IAsyncEnumerableConverterFactory : JsonConverterFactory
     {
         public override bool CanConvert(Type typeToConvert) => GetAsyncEnumerableInterface(typeToConvert) is not null;
 
+        [UnconditionalSuppressMessage("AOT", "IL3050:Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.", Justification = "<Pending>")]
```

Comment:
Unless we have guarantees that the typeToConvert and elementType are always reference type, or the code got compiled through other means, this is not safe to suppress.